executor/aggregate.go

```diff
@@ -128,6 +128,10 @@ func (e *HashAggExec) innerNext() (ret bool, err error) {
 		if srcRow == nil {
 			return false, nil
 		}
+		err = srcRow.DecodeValues(e.Src.Schema())
```

Comment:
There are lots of redundant code pattern like:
Next()
DecodeValues()

It's very easy to make a mistake. Maybe a better way is add a new interface like NextDecodedRow().  So we can keep our code clean.